### PR TITLE
#157248512 Add Token Authentication

### PR DESCRIPTION
**Pull request by ghost**
#### What does this PR do?
* This pull will update all endpoints to work with token authentication
#### Description of Task to be completed?
* Added `decorator.py` used to create the `token_required decorator
* Added `token Authentication` on all endpoints
* Modified `test cases` to include `tokens` in their request `headers`
* Modified `yaml files` to include `tokens field`.
#### How should this be manually tested?
* After setting  up the application as described in `README.md` and running **`python3 run.py`**, the endpoints can be tested from the swagger interface. The link is below.
[Documentation and testing](http://localhost:5000/apidocs/) 
#### What are the relevant pivotal tracker stories?
* [#157248512](https://www.pivotaltracker.com/story/show/157248512)

### CODE REVIEW

**Comment by RrNn** [comment]
-Remember to add the .pytest_cache/v/cache directory in your .git ignore file. It is in the tests folder. I think it comes after running your tests
-the .coveralls.yml file is not needed, only the .travis.yml file will be required for your TravisCI integration and Coverall test coverage

**Comment by chadwalt** [comment]
Fix the title for the pull request.

**Comment by chadwalt** [comment]
Fix the body of the pull request.

**Comment by chadwalt** [comment]
Change the title  to `#157248512 Add Token Authentication`

File: app/v1/models/models.py [hunk 1]
```diff
@@ -0,0 +1,263 @@
+import datetime
+# =====================================DbUsers==============================================
```
Comment (chadwalt) [inline app/v1/models/models.py:2]: Get rid of this. Just put some description of what the class should do. For example

`## Create Users .....`

File: app/v1/models/models.py [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+    def __init__(self):
+        self.all_users = dict()
+        self.all_emails = list()
+        self.id = 1
+
+    def add_user(self, email, username, password, address):
+        if email not in self.all_emails:
+            try:
+                self.all_users[username]
+            except KeyError:
```
Comment (chadwalt) [inline app/v1/models/models.py:14]: I thought we talked about this. The except should be reporting errors. But I see you still have actual code implemented there.

File: app/v1/models/models.py [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+    def __init__(self):
+        super().__init__()
+
+    def add_user(self, email, username, password, address, category='caterer', brand_name=''):
+        if email not in self.all_emails:
+            try:
+                self.all_users[username]
+            except KeyError:
+                brand_name_final = brand_name if brand_name else username
+                self.all_users[username] = dict(email=email, username=username, password=password, address=address,
```
Comment (chadwalt) [inline app/v1/models/models.py:55]: Avoid this. Report errors do not implement actual code.

File: app/v1/models/models.py [hunk 4]
```diff
@@ -0,0 +1,263 @@
...
+class DbMeals:
+    def __init__(self):
+        self.meals = dict()
+        # self.id
+
+    def add_meal(self, caterer, meal_name, price):
+        try:
+            self.meals[caterer]
+        except KeyError:
+            meals = list()
```
Comment (chadwalt) [inline app/v1/models/models.py:73]: the same here, report errors.

File: app/v1/models/models.py [hunk 5]
```diff
@@ -0,0 +1,263 @@
...
+        if isinstance(daily_menu, list):
+            self.menu[caterer] = daily_menu
+            return True
+        return False
+
+    def get_menu(self):
+        return self.menu
+
+
+# ===========================================DbOrders===============================
```
Comment (chadwalt) [inline app/v1/models/models.py:157]: Get rid of this

File: app/v1/views/meals.py [hunk 6]
```diff
@@ -10,10 +11,11 @@
 
 @meals.route('/meals/', methods=['POST'])
```
Comment (chadwalt) [inline app/v1/views/meals.py:12]: Add a comment and tell us what the route is for

File: app/v1/views/meals.py [hunk 7]
```diff
@@ -10,10 +11,11 @@
...
 @swag_from('api_doc/create_meals.yml')
-def create_meal():
+@token_required(admin=True)
+def create_meal(current_user):
     data = request.get_json()
-
-    if meals_db.add_meal(data['username'], data['name'], data['price']):
+    username = current_user[1]
+    if meals_db.add_meal(username, data['name'], data['price']):
```
Comment (chadwalt) [inline app/v1/views/meals.py:18]: Verify data before you store it

**Review by chadwalt — CHANGES_REQUESTED** [review]

**Pull request merged**